lms/events/event.py

```diff
@@ -34,60 +31,52 @@ class BaseEvent:  # pylint:disable=too-many-instance-attributes
     data: dict | None = None
     """Extra data to associate with this event"""
 
-    def __post_init__(self):
-        for event_field in fields(self):
-            getter_name = f"_get_{event_field.name}"
-
-            # If we don't have a value for `field` and we have implemented
-            # a _get_`field.name` method, use that to get a default
-            if not getattr(self, event_field.name) and hasattr(self, getter_name):
-                setattr(self, event_field.name, getattr(self, getter_name)())
+    Type = EventType.Type
+    """Expose the type here for the callers convenience"""
 
     def serialize(self) -> dict:
         return {
             field.name: getattr(self, field.name)
             for field in fields(self)
             # Excluded non-serializable fields
-            if field.name not in ["request"]
+            if field.name not in ["request", "Type"]
         }
 
 
 @dataclass
 class LTIEvent(BaseEvent):
-    """
-    Class to represent LTI-related events.
-
-    All the _get_`field`  method are used by the base class to
-    fill up any missing fields
-    """
-
-    def _get_user_id(self):
-        return self.request.user.id
-
-    # These methods provide defaults for each field and are used in
-    # `BaseEvent.__post_init__` above.
-    def _get_role_ids(self):
-        return [role.id for role in self.request.lti_user.lti_roles]
-
-    def _get_application_instance_id(self):
-        return self.request.lti_user.application_instance_id
-
-    def _get_course_id(self):
-        if course := self.request.find_service(name="course").get_by_context_id(
-            self.request.lti_user.lti.course_id
+    """Class to represent LTI-related events."""
+
+    @classmethod
+    def from_request(
+        cls, request: Request, type_: EventType.Type, data: dict | None = None
```

Comment:
Why `type_` instead of `type`?

Reply:
type is a reserved word so that would mask it. It will work unless you actually wanted to use the original type, but pylint will complain.